packages/frontity/src/utils/slot-and-fill/use-fills.tsx

```diff
@@ -0,0 +1,75 @@
+import get from "get-value";
+import { useConnect, warn } from "../..";
+import { Fill as StateFill, Package } from "../../../types";
+
+/**
+ * The fill items returned by the useFills hooks. They contain the same
+ * information as {@link StateFill} plus the final React component and a key
+ * to iterate over the array.
+ */
+interface Fill extends StateFill {
+  /**
+   * The React component that needs to be injected in the Slot for this fill.
+   */
+  Fill: React.ReactType;
+
+  /**
+   * A unique key that can be used when iterating over the array in React.
+   */
+  key: string;
+}
+
+/**
+ * A React hook to ease the creation of `Slot` components.
+ *
+ * @param name The name of the Slot that you want to fill.
+ *
+ * @return Array of fill objects that you can use to inject in the slot for this name.
+ */
+const useFills = (name: string): Fill[] => {
+  const { state, libraries } = useConnect<Package>();
+
+  if (!name) {
+    warn("You should pass the name of the slot that you would like to fill!");
+    return [];
+  }
+
+  return (
+    // Flat all the fills and turn them into entries.
+    Object.values(state.fills || {})
+      .reduce<[string, StateFill][]>(
+        (fills, fill) => [...fills, ...Object.entries(fill)],
```

Comment:
Just a nit but wouldn't this be clearer as:
```
 (fills, namespace) => [...fills, ...Object.entries(namespace)],
```

Reply:
Oh, absolutely. I'll change it.